File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraRichEdit;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
                private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { MyTextProvider.GetHTMLText(i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }


        public Form1()
        {
            InitializeComponent();
            gridControl1.DataSource = CreateTable(20);
            RepositoryItemRichTextEdit ri = new RepositoryItemRichTextEdit();
            ri.DocumentFormat = DocumentFormat.Html;
            gridView1.Columns[0].ColumnEdit = ri;
            gridView1.OptionsView.RowAutoHeight = true;

        }
    }

    public static class MyTextProvider
    {
        private const string sourceText = @"<b><font color=#00FFAA>Name<font color=#FF0011>_Number_</font></font></b><br>AddressLine1<br>AddressLine2<br>";
        public static string GetHTMLText(int index)
        {
            return sourceText.Replace("_Number_", index.ToString());
        }
    }

}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraRichEdit

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
				Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { MyTextProvider.GetHTMLText(i), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
				End Function


		Public Sub New()
			InitializeComponent()
			gridControl1.DataSource = CreateTable(20)
			Dim ri As New RepositoryItemRichTextEdit()
			ri.DocumentFormat = DocumentFormat.Html
			gridView1.Columns(0).ColumnEdit = ri
			gridView1.OptionsView.RowAutoHeight = True

		End Sub
	End Class

	Public NotInheritable Class MyTextProvider
		Private Const sourceText As String = "<b><font color=#00FFAA>Name<font color=#FF0011>_Number_</font></font></b><br>AddressLine1<br>AddressLine2<br>"
		Private Sub New()
		End Sub
		Public Shared Function GetHTMLText(ByVal index As Integer) As String
			Return sourceText.Replace("_Number_", index.ToString())
		End Function
	End Class

End Namespace